Source/Scene/Cesium3DTileStyle.js

```diff
@@ -159,16 +159,29 @@ define([
         that._ready = true;
     }
 
-    function getExpression(tileStyle, value) {
+    function getExpression(tileStyle, value, key) {
```

Comment:
Maybe it would be better to separate getting the expression from setting the json. One idea:

```js
function getJsonFromExpression(expression) {
    if (defined(value.expression)) {
        return value.expression;
    } else if (defined(value.conditionsExpression)) {
        return clone(value.conditionsExpression, true);
    }
    return expression;
}
```

```js
set : function(value) {
    this._show = getExpression(this, value);
    this._style.show = getJsonFromExpression(this._show);
}
```

A couple things to handle are:
* `this._style` would need to be initialized to an empty object
* What happens if the expression is a custom expression? Should that be added to the json as is? Or should `getJsonFromExpression` return undefined in that case.

Reply:
Thanks for the feedback @lilleyse. True, _abusing_ `getExpression` in this fashion is not very intuitive, I like your proposal and will implement something along those lines.

1. `this._style` initialization in the constructor makes sense to me, but this would change a non-experimental API, as the default value for `.style` would not longer be `undefined` but `Object`, or would you still return `undefined` in the getter, if the Object is empty?
2. Haven't thought of custom expressions tbh. Both approaches (`undefined` or the expression instance) have their uses. You cant serialize the expression, but you could use it to clone the style at runtime. I would lean toward an implementation that is closest to the current behavior, I'll have to checkout what that is first, but would assume it to be `undefined`?

Reply:
Changed my mind on 2.. Behavior as is now adds the custom expression passed in the options to `style`, this should remain unchanged. Added some specs to outline the behavior.